src/pyfmi/fmi_util.pyx

```diff
@@ -944,6 +1020,23 @@ cdef class DumpData:
             if self.bool_size > 0:
                 self.model_me2._get_boolean(self.bool_var_ref, self.bool_size, self.bool_var_tmp)
                 self.dump_data(self.bool_var_tmp)
+        elif self.model_me3_instance:
+
+            self.time_tmp[0] = self.model_me3.time
+            self.dump_data(self.time_tmp)
+
+            if self.real_size > 0:
+                # TODO: Do we want a get_float64_by_list also for performance reasons?
+                r = self.model_me3.get_float64(self.real_var_ref)
```

Comment:
Don't we need a separation of real_var_ref into float32, 64?

Reply:
I think you are right and same for `int32|64`, I will fix that tomorrow.

Reply:
@PeterMeisrimelModelon a lot of changes in commit https://github.com/modelon-community/PyFMI/pull/309/commits/46c479850f24e90480ff246216031f6a44efa905